lib/cryptoexchange/client.rb

```diff
@@ -24,6 +26,41 @@ def ticker(market_pair)
           t.base == market_pair.base && t.target == market_pair.target
         end
       end
+    rescue HTTP::ConnectionError, HTTP::TimeoutError, JSON::ParserError
+      return { error: 'exchange service temporary unavailable' }
+    end
+
+    def available_exchanges
+      folder_names = Dir[File.join(File.dirname(__dir__), 'cryptoexchange', 'exchanges', '**')]
+      folder_names.map do |folder_name|
+        folder_name.split('/').last
+      end
+    end
+
+    def exchange_for(currency)
+      exchanges = {
+        base: [],
+        target: [],
+        temporary_unavailable: []
+      }
+      available_exchanges.each do |exchange|
+        pairs = pairs(exchange)
+        if pairs.is_a?(Hash)
+          exchanges[:temporary_unavailable].push(exchange)
+          next
+        else
+          pairs(exchange).each do |pair|
+            if pair.base == currency.upcase
+              exchanges[:base].push(exchange)
+              break
+            elsif pair.target == currency.upcase
+              exchanges[:target].push(exchange)
+              break
+            end
+          end
+        end
+      end
+      exchanges
```

Comment:
@Swingcloud actually let's just return it as one array and ignore the ones that raises a HTTP error

Reply:
will do